cuDSS/simple_matrix_market/matrix_market_reader.h

```diff
@@ -0,0 +1,197 @@
+#include <fstream>
+#include <iostream>
+#include <sstream>
+#include <string>
+#include <algorithm>
+
+enum MtxReaderStatus {
+    MtxReaderSuccess,
+    MtxReaderErrorFileNotFound,
+    MtxReaderErrorFileMemAllocFailed,
+    MtxReaderErrorWrongNnz,
+    MtxReaderErrorUpperViewButLowerFound,
+    MtxReaderErrorLowerViewButUpperFound,
+    MtxReaderErrorOutOfBoundRowIndex,
+    MtxReaderErrorOfBoundColIndex
+
+};
+
+template <typename _TYPE_>
+int matrix_reader(std::string filename, int& n, int& nnz, int** csr_offsets_h,
+                  int** csr_columns_h, _TYPE_** csr_values_h, cudssMatrixViewType_t mview, bool verbose=true, bool verbose_err = true)
```

Comment:
In C and C++, names with an underscore, followed by an uppercase letter are reserved. I would suggest to name it `value_type` as that is more descriptive (and more in-line with our other examples).